app/helpers/application_helper/button/cloud_volume_new.rb

```diff
@@ -0,0 +1,18 @@
+class ApplicationHelper::Button::CloudVolumeNew < ApplicationHelper::Button::Basic
+  def calculate_properties
+    super
+    if disabled?
+      self[:title] = _("No cloud providers support creating cloud volumes.")
+    end
+  end
+
+  def disabled?
+    ems_clouds = EmsCloud.all
+    # if any connected provider supports this action,
+    # we can enable the button.
+    ems_clouds.each do |ems|
+      return false if CloudVolume.class_by_ems(ems).supports_create?
+    end
+    true
```

Comment:
I agree with you milan.
I appreciate the intention here and agree with it.
But we will bring back every (cloud) ems just to know if we should display a single button.

I thought a mechanism like rbac would be more useful here.
(yes, I do understand the difference. this is based upon existing ems not potential ems)

Unfortunately, I see this same code in [`cloud_network_controller`](https://github.com/ManageIQ/manageiq/blob/master/app/controllers/cloud_network_controller.rb#L78-L79) and [`cloud_subnet_controller`](https://github.com/ManageIQ/manageiq/blob/master/app/controllers/cloud_subnet_controller.rb#L76-L77).